### PR TITLE
gh-2148 - functions to convert elements and properties to tuples

### PR DESCRIPTION
**Pull request by p013570**

### CODE REVIEW

File: core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ToElementTuple.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017-2019 Crown Copyright
```
Comment (m55624) [inline core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ToElementTuple.java]: 2019

File: core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ToPropertiesTuple.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017-2019 Crown Copyright
```
Comment (m55624) [inline core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ToPropertiesTuple.java]: 2019

File: core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/ToPropertiesTupleTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017-2019 Crown Copyright
```
Comment (m55624) [inline core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/ToPropertiesTupleTest.java]: 2019

File: core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/ToElementTupleTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017-2019 Crown Copyright
```
Comment (m55624) [inline core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/ToElementTupleTest.java]: 2019

File: core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ToElementTuple.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+import static java.util.Objects.isNull;
+
+/**
+ * A {@code ToElementTuple} is a {@link KorypheFunction} that converts an {@link Element} into an {@link ElementTuple}.
+ */
+@Since("1.9.0")
```
Comment (d47853) [inline core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ToElementTuple.java]: This tag needs updating to 1.10.0

File: core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ToPropertiesTuple.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+import static java.util.Objects.isNull;
+
+/**
+ * A {@code ToPropertiesTuple} is a {@link KorypheFunction} that converts Element {@link Properties} into a {@link PropertiesTuple}.
+ */
+@Since("1.9.0")
```
Comment (d47853) [inline core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ToPropertiesTuple.java]: Likewise

**Pull request merged**